lms/services/roster.py

```diff
@@ -373,21 +504,105 @@ def _get_roster_roles(self, roster) -> list[LTIRole]:
         roles = {role for member in roster for role in member["roles"]}
         return self._lti_role_service.get_roles(list(roles))
 
-    def _get_lti_registration(self, lms_course) -> LTIRegistration:
-        lti_registration = self._db.scalars(
-            select(LTIRegistration)
-            .join(ApplicationInstance)
-            .where(LMSCourseApplicationInstance.lms_course_id == lms_course.id)
+    def _get_application_instance(self, lms_course) -> ApplicationInstance:
```

Comment:
Refactor the existing get_lti_registration to use the new get_application_instance.